### PR TITLE
[VTK 8.2.0/HDF5 1.10.5] Upgrade VTK and HDF5

### PR DESCRIPTION
**Pull request by Neumann-A**
Ran out of time to test the build but I think it is a good start.

### CODE REVIEW

**Comment by Rastaban** [comment]
fixes #5462

**Comment by Rastaban** [comment]
Thanks for taking this on! Here are the results I currently see in the CI system.  We are currently seeing flakeyness in our CI system with all ports that depend on qt5-base due to the way it caches its results, so if you are not able to reproduce the osg-qt failures locally they can probably be ignored for now.  Let me know if you would like me to attach any of the failure logs.

```
 Processing arm-uwp                        411 vs 411
 Processing arm64-windows                  470 vs 470
 Processing x64-linux                      606 vs 610
     alembic                                                     Skip vs Fail
     hdf5                          **regression**                Fail vs Pass
     highfive                                                    Skip vs Fail
     itk                                                         Skip vs Fail
     kealib                                                      Skip vs Pass
     matio                                                       Skip vs Fail
     netcdf-c                                                    Skip vs Pass
     netcdf-cxx4                                                 Skip vs Pass
     vtk                                                         Skip vs Fail
 Processing x64-osx                        597 vs 604
     alembic                                                     Skip vs Fail
     hdf5                          **regression**                Fail vs Pass
     highfive                                                    Skip vs Pass
     itk                                                         Skip vs Fail
     kealib                                                      Skip vs Pass
     matio                                                       Skip vs Fail
     netcdf-c                                                    Skip vs Pass
     netcdf-cxx4                                                 Skip vs Pass
     vtk                                                         Skip vs Pass
     vtk-dicom                                                   Skip vs Pass
 Processing x64-uwp                        439 vs 439
 Processing x64-windows                    877 vs 881
     osg-qt                        **regression**                Fail vs Pass
     pcl                                                         Skip vs Pass
     vtk                           **regression**                Fail vs Pass
     vtk-dicom                                                   Skip vs Pass
 Processing x64-windows-static             773 vs 776
     pcl                                                         Skip vs Pass
     vtk                           **regression**                Fail vs Pass
     vtk-dicom                                                   Skip vs Pass
 Processing x86-windows                    856 vs 860
     osg-qt                        **regression**                Fail vs Pass
     pcl                                                         Skip vs Pass
     vtk                           **regression**                Fail vs Pass
     vtk-dicom                                                   Skip vs Pass
```

**Comment by Neumann-A** [comment]
Current problem: Get VTK to link ogg (or Ogg? maybe a name conflict?)

**Comment by Neumann-A** [comment]
@Rastaban: I need to know where the regressions are after the last commit and what happend. I succesfully build x64-windows but times up for today.

**Comment by Neumann-A** [comment]
ok i'll give up with trying to test pcl with vtk 8.2.0 on linux. 
Hitting the following while compiling with gcc 7.3 flann: 
 `c++: internal compiler error: Killed (program cc1plus). ` 

So on x64-windows PCL builds fine. On linux vtk 8.2 builds fine. Currently testing x64-windows-static but I don't expect any surprises. So if there are still regression please let me know and attach the logs.

**Comment by Neumann-A** [comment]
I spoke too soon but the last commit fixed the x64-windows-static build. On my machine it build successfully but failed to copy to installed/x64-windows-static because I ran out of disk space.
Nevertheless it reported: `Installing package vtk[core]:x64-windows-static... done`
So this PR should be wrapped up... lets wait what CI tells us

**Comment by Rastaban** [comment]
Sorry for the delay, here is what I see now.  As mentioned earlier, any failure that has a dependancy on qt5-base can be ignored for now because of #5566.  Let me know if you need any of the failure logs.

```
processing arm-uwp                        410 vs 411
     qhull                         **regression**                Fail vs Pass
 Processing arm64-windows                  470 vs 471
     qhull                         **regression**                Fail vs Pass
 Processing x64-linux                      615 vs 614
     highfive                                                    Pass vs Fail
     kealib                        **regression**                Fail vs Pass
     qhull                                                       Pass vs Fail
 Processing x64-osx                        609 vs 607
     pcl                                                         Pass vs Skip
     qhull                                                       Pass vs Fail
 Processing x64-uwp                        438 vs 439
     qhull                         **regression**                Fail vs Pass
 Processing x64-windows                    881 vs 884
     osg-qt                        **regression**                Fail vs Pass
     pcl                                                         Skip vs Pass
     qhull                         **regression**                Fail vs Pass
 Processing x64-windows-static             774 vs 778
     pcl                                                         Skip vs Pass
     qhull                         **regression**                Fail vs Pass
     vtk                           **regression**                Fail vs Pass
     vtk-dicom                                                   Skip vs Pass
 Processing x86-windows                    861 vs 864
     osg-qt                        **regression**                Fail vs Pass
     pcl                                                         Skip vs Pass
     qhull                         **regression**                Fail vs Pass

```

**Comment by Neumann-A** [comment]
> kealib                        **regression**                Fail vs Pass

Error:
```
/vcpkg/buildtrees/kealib/src/kealib-1.4.7/trunk/include/libkea/KEAAttributeTableInMem.h:38:10: fatal error: H5Cpp.h: No such file or directory
 #include "H5Cpp.h"
```
Can somebody report upstream that they need to use #include <H5Cpp.h> everywhere to correctly include this header? Maybe: @miurahr?

**Comment by Neumann-A** [comment]
```
 Processing x64-windows-static             774 vs 778
     pcl                                                         Skip vs Pass
     qhull                         **regression**                Fail vs Pass
     vtk                           **regression**                Fail vs Pass
     vtk-dicom                                                   Skip vs Pass
```

@Rastaban: I cannot reproduce the vtk regression in x64-windows-static
```
Building package vtk[core]:x64-windows-static...
-- Downloading https://github.com/Kitware/VTK/archive/v8.2.0.tar.gz...
-- Extracting source D:/vcpkg_test/vcpkg/downloads/Kitware-VTK-v8.2.0.tar.gz
-- Applying patch fix-find-lz4.patch
-- Applying patch fix_ogg_linkage.patch
-- Using source at D:/vcpkg_test/vcpkg/buildtrees/vtk/src/v8.2.0-4dfb703163
-- Configuring x64-windows-static
-- Building x64-windows-static-dbg
-- Building x64-windows-static-rel
-- Performing post-build validation
-- Performing post-build validation done
Building package vtk[core]:x64-windows-static... done
Installing package vtk[core]:x64-windows-static...
Installing package vtk[core]:x64-windows-static... done
Elapsed time for package vtk:x64-windows-static: 20.81 min
```

**Comment by Neumann-A** [comment]
@Rastaban I would like to know the regression of https://github.com/Microsoft/vcpkg/pull/5574/commits/07b0183a779a66b2862c97ab2a5f3f8338a8d775 and if https://github.com/Microsoft/vcpkg/pull/5574/commits/af06207c3d7f797be8d69fa511c2f6dcfe681fbe introduced any new regressions.
If https://github.com/Microsoft/vcpkg/pull/5574/commits/07b0183a779a66b2862c97ab2a5f3f8338a8d775 still has the regressions in x64-windows-static then I also need the log files because I couldn't reproduce those.

**Comment by PhoebeHui** [comment]
@Neumann-A, check the latest results and attach the failure logs:
```
 Processing arm-uwp                        412 vs 412
 Processing arm64-windows                  472 vs 472
 Processing x64-linux                      619 vs 617
     highfive                                                    Pass vs Fail
     qhull                                                       Pass vs Fail
 Processing x64-osx                        612 vs 610
     pcl                                                         Pass vs Skip
     qhull                                                       Pass vs Fail
 Processing x64-uwp                        440 vs 440
 Processing x64-windows                    878 vs 886
     gdal                                                        Skip vs Pass
     netcdf-c                      **regression**                Fail vs Pass
     netcdf-cxx4                                                 Skip vs Pass
     osg                                                         Skip vs Pass
     osg-qt                                                      Skip vs Pass
     osgearth                                                    Skip vs Pass
     pdal                                                        Skip vs Pass
     pdal-c                                                      Skip vs Pass
 Processing x64-windows-static             778 vs 780
     pcl                                                         Skip vs Pass
     qhull                         **regression**                Fail vs Pass
 Processing x86-windows                    858 vs 866
     gdal                                                        Skip vs Pass
     netcdf-c                      **regression**                Fail vs Pass
     netcdf-cxx4                                                 Skip vs Pass
     osg                                                         Skip vs Pass
     osg-qt                                                      Skip vs Pass
     osgearth                                                    Skip vs Pass
     pdal                                                        Skip vs Pass
     pdal-c                                                      Skip vs Pass
```
[failureLogs.zip](https://github.com/Microsoft/vcpkg/files/2956045/failureLogs.zip)

**Comment by Neumann-A** [comment]
@PhoebeHui 
The qhull failure raises my eyebrows:
`LINK : fatal error LNK1201: error writing to program database 'C:\vsts\_work\2\s\buildtrees\qhull\x64-windows-static-rel\qhull.pdb'; check for insufficient disk space, invalid path, or insufficient privilege`
But I can also reproduce it locally with the same error message. Any idea where that might come from? I see that the folder in windows is partially write protected for some strange reason.

**Comment by Rastaban** [comment]
In the latest test run the only regressions are qhull on arm64-windows and x86-windows with the same `LNK1201` error

**Comment by Neumann-A** [comment]
@Rastaban: Hmm tried it at home for x86-windows and it worked the first time. Removed it and updated Visual Studio 2017 to newest version -> LNK 1201. Seems to be a regression in the build tools not correctly releasing the lock on the file or something similar.

**Comment by Neumann-A** [comment]
and to top it off.... if I readd `PREFER_NINJA` it works again o.O. How should we treat that? Certainly not a vcpkg issue

**Comment by miurahr** [comment]
> Can somebody report upstream that they need to use #include <H5Cpp.h> everywhere to correctly include this header? Maybe: @miurahr?

[This commit](https://bitbucket.org/chchrsc/kealib/commits/bf615ef37f9baf66b505243fc34e80914c54155e) maybe by @gillins

**Comment by gillins** [comment]
Is the HDF5 C++ library still being built? FYI might be worth upgrading to `kealib` 1.4.11 - many improvements to the build system have been made.

**Comment by Rastaban** [comment]
@Neumann-A you may be right about the toolset regression.  Looking through the internal MSVC bug database there have been a few issues related to LNK1201 that have been fixed.

**Comment by Rastaban** [comment]
Other than the qhull failure on x64-windows-static everything else passes the tests.  Are there any other outstanding issue that need to be resolved before this is merged?

**Comment by Neumann-A** [comment]
@Rastaban: should i clean up the commit history or do you want to merge it the way it is? There are probably other portfiles which can remove extra code changing *targets.cmake due to commit https://github.com/Microsoft/vcpkg/pull/5574/commits/0b5a5767d066cc3f84598435d46400555acb0e77 but that could be done in another PR. The problem was that the release version of the targets file was missing the debug information due to the fact that cmake was simply not finding the debug versions due to the missing path. (folly is such an example I at least know of.)

File: ports/qhull/portfile.cmake [hunk 1]
```diff
@@ -7,7 +7,20 @@ vcpkg_from_github(
     SHA512 8f5177ea45f82fa28f13e95105497e7e29086d7301e1cb8d3860fff09ebf8d0f01cfcb0f044c422f0ac0ba94b845bba223232e5eeb613bf671f65a569b8766d0
     HEAD_REF master
 )
-
+if(${VCPKG_TARGET_TRIPLET} STREQUAL "x64-windows-static")
```
Comment (Rastaban) [inline ports/qhull/portfile.cmake]: this should be `${TARGET_TRIPLET}`

**Comment by Rastaban** [comment]
I was going to do a Squash and Merge, but if you are willing to cleanup the commit history then I can just merge it when you are done.

File: ports/qhull/portfile.cmake (hunk 1 above)
Comment (Neumann-A) [reply ports/qhull/portfile.cmake]: fixed in 
1feb0dc

**Pull request merged**